cmd/goal/application.go

```diff
@@ -266,6 +306,18 @@ func processAppInputFile() (args [][]byte, accounts []string, foreignApps []uint
 	return parseAppInputs(inputs)
 }
 
+func splitAppArgsByComma() {
```

Comment:
I will write a docstring for that, but I will anyways explain here first.

The name is not correct, but it was first made out of purpose of splitting input string like form `str:a,str:b`, since the parsing of `appArgs` uses `StringSliceVar` and cut the string into slices with comma.

I switched it to `StringArrayVar` to support parsing of `abi:...` string, or it will cut the string into pieces. But unfortunately `StringArrayVar` will allow empty string parsed into `appArgs`.

In this way we have to first filter out the empty string with a not good magic way. If we do not do so, then it will contradict with some behavior tests...

What is your thoughts on that?

Reply:
It's ok to filter out the blanks, it just needs to be named appropriately.  And it should take a slice and return a slice, not just modify globals.